pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -333,3 +333,12 @@ def predict_step(self, *args, **kwargs):
     def post_training_step(self):
         if not self.lightning_module.automatic_optimization:
             self.model.require_backward_grad_sync = True
+
+    @classmethod
+    def register_plugins(cls, plugin_registry: Dict) -> None:
+        plugin_registry.register(
+            "ddp_find_unused_parameters_false",
+            cls,
+            description="DDP Plugin with `find_unused_parameters` as False",
+            find_unused_parameters=False
+        )
```

Comment:
since it's now easy to toggle this on and off, why don't we keep the pytorch default and for users who run into this failure, they can specify this plugin on their trainer? this will give a performance improvement for a wide number of users (saves them time/$) and keeps us in sync with pytorch. could that be a follow up to this PR?

Reply:
Valid point!
As I don't know the exact reasons why we switched the default. @SeanNaren Your opinions?

Reply:
I fully agree with you here but am concerned with this variable being turned on and off by default so much, as this could be frustrating for users :(

Also if we do decide to revert back to default of False since we have the plugin registry, we should catch the exception thrown when `find_unused_parameters` is required if possible and add a message suggesting `ddp_find_unused_parameters_true`.

I'd say regardless of the decision, we'll make that a followup!

Reply:
Hey @ananthsub. 

I would strongly prefer to keep it turned `ON` for now. 
Lot of users use DDP without knowing the intricacies between it and find themselves with hanging code due to  `find_unused_parameters=False`. When we turned this off to follow PyTorch default, we got a burst of bugs related to it.
Better to keep it in the performance section and easy turn on with @kaushikb11 addition.

T.C

Reply:
@tchaton I made https://github.com/PyTorchLightning/pytorch-lightning/issues/7330 - let's follow up there